JavaScript/3-dequeue.js

```diff
@@ -4,6 +4,7 @@ class Dequeue {
   constructor() {
     this.first = null;
     this.last = null;
+    this._emitter = new (require('events'));    
```

Comment:
Don't use `require` inside constructor, this is obvious memory leak, you can move it to first lines of this file.